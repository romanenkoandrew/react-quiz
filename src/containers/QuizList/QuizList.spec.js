import React from "react";
import { shallow } from "enzyme";
import { QuizList } from "./QuizList";
import QuizListConnect from "./QuizList";
import configureStore from "redux-mock-store";

const mockStore = configureStore();

describe("QuizList component", () => {
  let component, fetchQuizes;
  beforeEach(() => {
    fetchQuizes = jest.fn();
    component = shallow(
      <QuizList
        quizes={[
          { id: 1, name: "name" },
          { id: 2, name: "name2" },
        ]}
        fetchQuizes={fetchQuizes}
      />
    );
  });
  it("should render QuizList with quizes", () => {
    expect(component).toMatchSnapshot();
  });
  it("should render QuizList with quizes and with loading", () => {
    const component2 = shallow(
      <QuizList
        quizes={[
          { id: 1, name: "name" },
          { id: 2, name: "name2" },
        ]}
        fetchQuizes={fetchQuizes}
        loading
      />
    );
    expect(component2).toMatchSnapshot();
  });
  it("should render QuizList without quizes", () => {
    const component3 = shallow(
      <QuizList quizes={[]} fetchQuizes={fetchQuizes} />
    );
    expect(component3).toMatchSnapshot();
  });
  it("should quizes and loading in QuizList props, calls fetchQuizes", () => {
    const store = mockStore({
      quiz: {
        quizes: [
          { id: 1, name: "name" },
          { id: 2, name: "name2" },
        ],
        loading: true,
      },
    });
    const component4 = shallow(<QuizListConnect store={store} />);
    expect(component4.children().props().quizes).toBeTruthy();
    expect(component4.children().props().loading).toBeTruthy();
  });
});
