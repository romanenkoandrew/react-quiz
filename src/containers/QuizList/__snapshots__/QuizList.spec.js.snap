// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizList component should render QuizList with quizes 1`] = `
<div
  className="QuizList"
>
  <div>
    <h1>
      Список тестов
    </h1>
    <ul>
      <li
        key="1"
      >
        <NavLink
          to="/quiz/1"
        >
          name
        </NavLink>
      </li>
      <li
        key="2"
      >
        <NavLink
          to="/quiz/2"
        >
          name2
        </NavLink>
      </li>
    </ul>
  </div>
</div>
`;

exports[`QuizList component should render QuizList with quizes and with loading 1`] = `
<div
  className="QuizList"
>
  <div>
    <h1>
      Список тестов
    </h1>
    <Loader />
  </div>
</div>
`;

exports[`QuizList component should render QuizList without quizes 1`] = `
<div
  className="QuizList"
>
  <div>
    <h1>
      Список тестов
    </h1>
    <ul />
  </div>
</div>
`;
